src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -922,6 +924,39 @@ reconnectServerClients :: ProtocolServerClient v err msg => AgentClient -> (Agen
 reconnectServerClients c clientsSel =
   readTVarIO (clientsSel c) >>= mapM_ (forkIO . closeClient_ c)
 
+reconnectSMPServerClients :: AgentClient -> AM' ()
+reconnectSMPServerClients c = do
+  -- 1. swap smpClients to empty map, move active subscriptions to pending
+  (clients, prevActive) <- atomically $ do
+    clients <- smpClients c `swapTVar` M.empty
+    prevActive <- activeToPendingQueues (activeSubs c) (pendingSubs c)
+    pure (clients, prevActive)
+  -- 2. notify DOWN for connections that had active subscriptions
+  let downConns = groupConnsByServer prevActive
+  forM_ (M.toList downConns) $ \(server, connIds) ->
+    liftIO $ notifyDOWN server connIds
+  -- 3. close clients
+  mapM_ (liftIO . forkIO . closeClient_ c) clients
+  -- 4. resubscribe pending subscriptions
+  mode <- liftIO $ getSessionMode c
+  pending <- readTVarIO (getRcvQueues $ pendingSubs c)
+  -- Group transport sessions to avoid multiple UP events in case session mode is TSMUser
```

Comment:
not sure I understand it?

Reply:
is it maybe easier to get all connection IDs from active + pending, and then subscribe them?